### DISCUSSION TITLE
Do we need to deploy a new NFTDescriptor for V2.2?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#877] [General]
This is a question for you @smol-ninja and @andreivladbrg.

If we could avoid deploying the NFTDescriptor, we would save quite a bit of ETH in deployment costs.

Speaking of this, it might be helpful to write one or two migration tests showing that V2.2 can be integrated with the previous NFTDescriptor deployment.

**Comment by smol-ninja** [comment]
`NFTDescriptor` is backward compatible with the previous versions. However we've made the following changes to the NFTDescriptor:

- Adds [isTransferable](https://github.com/sablier-labs/v2-core/blob/76e29fee33befaeb54a82473c28cf182e6f8855d/src/SablierV2NFTDescriptor.sol#L96C1) value to NFT description.
- new NFT symbol: [SAB-V2-LOCKUP-TRA](https://github.com/sablier-labs/v2-core/blob/76e29fee33befaeb54a82473c28cf182e6f8855d/src/SablierV2NFTDescriptor.sol#L303C1-L304C38)

Do you still want to have tests written showing that its backward compatible?

**Reply by PaulRBerg** [reply]
Got it.In this case, we will have to make a new deployment.

And yes, we also want to write backward compatibility tests to show that the new NFT descriptor is compatible with the previous deployments

**Reply by smol-ninja** [reply]
Related: https://github.com/sablier-labs/v2-core/issues/881#issuecomment-2045369636

> While working on this, I came to realize the new NFTDescriptor is not compatible with v2.0 but only with v2.1. This is due to the fact the new NFTDescriptor includes `isTransferable` in the svg description. v2.0 doesn't have `isTransferable`.

https://github.com/sablier-labs/v2-core/blob/76e29fee33befaeb54a82473c28cf182e6f8855d/src/SablierV2NFTDescriptor.sol#L96
